messaging/jms/src/main/java/io/helidon/messaging/connectors/jms/JmsMessage.java

```diff
@@ -0,0 +1,243 @@
+        } else if (msg instanceof BytesMessage) {
+            return new JmsBytesMessage((BytesMessage) msg, executor, sessionEntry);
+        } else {
+            throw new MessagingException("Unsupported JMS message type");
+        }
+    }
+
+    /**
+     * Create a message with the given payload.
+     *
+     * @param msg JMS message to be wrapped
+     * @return A message with the given payload, and an ack function
+     */
+    static JmsMessage<?> of(javax.jms.Message msg) {
+        if (msg instanceof TextMessage) {
+            return JmsMessage.of((TextMessage) msg);
+        } else if (msg instanceof BytesMessage) {
+            return JmsMessage.of((BytesMessage) msg);
+        } else {
+            throw new MessagingException("Unsupported JMS message type");
+        }
+    }
+
+    /**
+     * Create a message with the given payload.
+     *
+     * @param msg JMS message to be wrapped
+     * @return A message with the given payload, and an ack function
+     */
+    static JmsMessage<String> of(javax.jms.TextMessage msg) {
+        return new JmsTextMessage(msg, Executors.newWorkStealingPool(), null);
+    }
+
+    /**
+     * Create a message with the given payload.
+     *
+     * @param msg JMS message to be wrapped
+     * @return A message with the given payload, and an ack function
+     */
+    static JmsMessage<InputStream> of(javax.jms.BytesMessage msg) {
+        return new JmsBytesMessage(msg, Executors.newWorkStealingPool(), null);
+    }
+
+    abstract class AbstractJmsMessage<T> implements JmsMessage<T> {
+
+        private final Executor executor;
+        private final SessionMetadata sharedSessionEntry;
+        private boolean acked = false;
+
+        protected AbstractJmsMessage(Executor executor, SessionMetadata sharedSessionEntry) {
+            this.sharedSessionEntry = sharedSessionEntry;
+            this.executor = executor;
+        }
+
+        @Override
+        public Session getJmsSession() {
+            return sharedSessionEntry.getSession();
+        }
+
+        @Override
+        public Connection getJmsConnection() {
+            return sharedSessionEntry.getConnection();
+        }
+
+        @Override
+        public ConnectionFactory getJmsConnectionFactory() {
+            return sharedSessionEntry.getConnectionFactory();
+        }
+
+        @Override
+        public boolean isAck() {
+            return acked;
+        }
+
+        @Override
+        public CompletionStage<Void> ack() {
+            return CompletableFuture.runAsync(() -> {
+                try {
+                    this.getJmsMessage().acknowledge();
+                    acked = true;
+                } catch (JMSException e) {
+                    LOGGER.log(Level.SEVERE, e, () -> "Error during acknowledgement of JMS message");
+                }
+            }, executor);
+        }
+
+    }
+
+    class JmsTextMessage extends AbstractJmsMessage<String> {
```

Comment:
Isn't this class automatically public, as it is declared on an interface? If so, missing javadoc

Reply:
Yes it is thx!